controller_interface/src/controller_interface.cpp

```diff
@@ -110,6 +110,11 @@ const rclcpp_lifecycle::State & ControllerInterface::get_state() const
   return node_->get_current_state();
 }
 
+void ControllerInterface::set_node(std::shared_ptr<rclcpp_lifecycle::LifecycleNode> node)
```

Comment:
Should we pass this by reference?

```suggestion
void ControllerInterface::set_node(std::shared_ptr<rclcpp_lifecycle::LifecycleNode> & node)
```

Reply:
Probably? I'll make that a const ref.